dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/ResourceLink.java

```diff
@@ -0,0 +1,165 @@
+package com.dotmarketing.portlets.contentlet.model;
+
+import com.dotcms.contenttype.model.type.FileAssetContentType;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.fileassets.business.FileAsset;
+import com.dotmarketing.portlets.fileassets.business.FileAssetAPI;
+import com.dotmarketing.util.InodeUtils;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+import javax.servlet.http.HttpServletRequest;
+
+/***
+ * This class is the result of a refactoring from an old JSP Snippet that was originally located in:
+ * `html/portlet/ext/contentlet/field/edit_field.jsp`
+ */
+public class ResourceLink {
+
+    static final String HOST_REQUEST_ATTRIBUTE = "host";
+
+    private static final String HTTP_PREFIX = "http://";
+
+    private static final String HTTPS_PREFIX = "https://";
+
+    private static final String LANG_ID_PARAM = "?language_id=";
+
+    private final String resourceLinkAsString;
+
+    private final String resourceLinkUriAsString;
+
+    private final String mimeType;
+
+    private final FileAsset fileAsset;
+
+    private final boolean editableAsText;
+
+    private final boolean downloadRestricted;
+
+    private ResourceLink(final String resourceLinkAsString, final String resourceLinkUriAsString,
+            final String mimeType,
+            final FileAsset fileAsset,
+            final boolean editableAsText,
+            final boolean downloadRestricted) {
+        this.resourceLinkAsString = resourceLinkAsString;
+        this.resourceLinkUriAsString = resourceLinkUriAsString;
+        this.mimeType = mimeType;
+        this.fileAsset = fileAsset;
+        this.editableAsText = editableAsText;
+        this.downloadRestricted = downloadRestricted;
+    }
+
+    public String getResourceLinkAsString() {
+        return resourceLinkAsString;
+    }
+
+    public String getResourceLinkUriAsString() {
+        return resourceLinkUriAsString;
+    }
+
+    public String getMimeType() {
+        return mimeType;
+    }
+
+    public FileAsset getFileAsset() {
+        return fileAsset;
+    }
+
+    public boolean isEditableAsText() {
+        return editableAsText;
+    }
+
+    public boolean isDownloadRestricted() {
+        return downloadRestricted;
+    }
+
+    @Override
+    public String toString() {
+        return resourceLinkAsString;
+    }
+
+    public static class ResourceLinkBuilder {
+
+        public final ResourceLink build(final HttpServletRequest request, final User user, final Contentlet contentlet) throws DotDataException, DotSecurityException {
```

Comment:
I am wondering if a ResourceLink could be created on non-web context (aka a Http request)
it could be a good idea to abstract the secure and host in parameters or one parameter, so that it could be call from a job for instance